File: Raspi/Freenove_RFID_Starter_Kit_for_Raspberry_Pi/Code/Python_GPIOZero_Code/04.1.1_BreathingLED/BreathingLED.py
#!/usr/bin/env python3
########################################################################
# Filename    : BreathingLED.py
# Description : Breathing LED
# Author      : www.freenove.com
# modification: 2023/05/11
########################################################################
from gpiozero import PWMLED
import time

led = PWMLED(18 ,initial_value=0 ,frequency=1000)
def loop():
    while True:
        for b in range(0, 101, 1):    # make the led brighter
            led.value = b / 100.0     # set dc value as the duty cycle
            time.sleep(0.01)
        time.sleep(1)
        for b in range(100, -1, -1):  # make the led darker
            led.value = b / 100.0     # set dc value as the duty cycle
            time.sleep(0.01)
        time.sleep(1)
def destroy():
    led.close()
if __name__ == '__main__':     # Program entrance
    print ('Program is starting ... ')
    try:
        loop()
    except KeyboardInterrupt:  # Press ctrl-c to end the program.
        destroy()
        print("Ending program")

File: Raspi/Freenove_RFID_Starter_Kit_for_Raspberry_Pi/Code/C_Code/04.1.1_BreathingLED/BreathingLED.c
/**********************************************************************
* Filename    : BreathingLED.c
* Description : Make breathing LED with PWM
* Author      : www.freenove.com
* modification: 2019/12/27
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>
#include <softPwm.h>

#define ledPin    1 

void main(void)
{
	int i;
	
	printf("Program is starting ... \n");
	
	wiringPiSetup();	//Initialize wiringPi.
	
	softPwmCreate(ledPin,  0, 100);//Creat SoftPWM pin
	
	while(1){
		for(i=0;i<100;i++){  //make the led brighter
			softPwmWrite(ledPin, i); 
			delay(20);
		}
		delay(300);
		for(i=100;i>=0;i--){  //make the led darker
			softPwmWrite(ledPin, i);
			delay(20);
		}
		delay(300);
	}
}

